prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -1226,15 +1226,20 @@ class Hl7Serializer(
      * @param truncationLimit the starting limit
      * @return the new truncation limit or starting limit if no special characters are found
      */
-    internal fun getTruncationLimitWithEncoding(value: String, truncationLimit: Int): Int {
-        val regex = "[&^~|]".toRegex()
-        val endIndex = min(value.length, truncationLimit)
-        val matchCount = regex.findAll(value.substring(0, endIndex)).count()
+    internal fun getTruncationLimitWithEncoding(value: String, truncationLimit: Int?): Int? {
 
-        return if (matchCount > 0) {
-            truncationLimit.minus(matchCount.times(2))
+        return if (truncationLimit != null) {
+            val regex = "[&^~]".toRegex()
```

Comment:
the bar ("|") is also a special character in HL7. Why was it removed from the regex list?

Reply:
Good catch, I'll add it. I probably deleted it by accident.